### PR TITLE
CT Redesign: Create/Update sandbox location search service #23654

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
https://github.com/department-of-veterans-affairs/va.gov-team/issues/23654

## Screenshots
Screen Shot 2021-05-13 at 1 01 52 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/gi-sandbox/constants.js [hunk 1]
```diff
@@ -20,6 +20,44 @@ export const KEY_CODES = Object.freeze({
...
+ * Mapbox api request countries
+ */
+
+export const CountriesList = ['us', 'pr', 'ph', 'gu', 'as', 'mp'];
+
+/**
+ * Mapbox api request types
+ */
+
+export const TypeList = ['place', 'region', 'postcode', 'locality'];
```
Comment (zurbergram) [inline src/applications/gi-sandbox/constants.js:44]: https://docs.mapbox.com/api/search/geocoding/#data-types

File: src/applications/gi-sandbox/actions/index.js [hunk 2]
```diff
@@ -27,12 +32,16 @@ export const FETCH_PROFILE_FAILED = 'FETCH_PROFILE_FAILED';
 export const FETCH_PROFILE_STARTED = 'FETCH_PROFILE_STARTED';
 export const FETCH_PROFILE_SUCCEEDED = 'FETCH_PROFILE_SUCCEEDED';
 export const FILTER_TOGGLED = 'FILTER_TOGGLED';
+export const GEOCODE_STARTED = 'GEOCODE_STARTED';
+export const GEOCODE_FAILED = 'GEOCODE_FAILED';
+export const GEOCODE_COMPLETE = 'GEOCODE_COMPLETE';
```
Comment (shawkey-daniel) [inline src/applications/gi-sandbox/actions/index.js:37]: We use SUCCEEDED for everything else

File: src/applications/gi-sandbox/constants.js [hunk 3]
```diff
@@ -20,6 +20,38 @@ export const KEY_CODES = Object.freeze({
...
 
+/**
+ * Mapbox init values
+ */
+export const MapboxInit = {
+  zoomInit: 3,
+  centerInit: {
+    lng: -99.27246093750001,
+    lat: 40.17887331434698,
+  },
```
Comment (shawkey-daniel) [inline src/applications/gi-sandbox/constants.js:31]: Why these magic init values?

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]

File: src/applications/gi-sandbox/constants.js (hunk 3 above)
Comment (zurbergram) [reply src/applications/gi-sandbox/constants.js:31]: Centerish of USA, lat/long is for Turkey Creek, NE
Used in https://github.com/department-of-veterans-affairs/vets-website/pull/17247
image

File: src/applications/gi-sandbox/actions/index.js (hunk 2 above)
Comment (zurbergram) [reply src/applications/gi-sandbox/actions/index.js]: changed

File: src/applications/gi-sandbox/constants.js (hunk 3 above)
Comment (shawkey-daniel) [reply src/applications/gi-sandbox/constants.js:31]: Can you create some constants, so we know why they were picked?

**Review by shawkey-daniel — APPROVED** [review]

File: src/applications/gi-sandbox/constants.js (hunk 3 above)
Comment (zurbergram) [reply src/applications/gi-sandbox/constants.js:33]: Added comment to explain lat/long and default zoom values

**Review by sonntag-adam — APPROVED** [review]

**Pull request merged**